### PR TITLE
Home Link: Render Home text if there is no attribute label present

### LINKED ISSUE #58307: Update defaults for Home Link block
### Description

It's now possible to use the Block Hooks API to insert inner blocks into the core Navigation block https://github.com/WordPress/gutenberg/pull/57754. Using this API currently doesn't allow you to set attributes along with the hooked block.

We found that when you use the API to insert the `core/home-link` block it has insufficient default attributes. When we insert this block via the API we can't set any attributes and one of the conditions for this block to render server-side is a label attribute, see [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/home-link/index.php#L131-L133).

When you insert the block via the editor, this gets added on the client-side. See [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/home-link/edit.js#L47-L54).

### Step-by-step reproduction instructions

1. Add the below code to your site.
2. Load frontend and see that the block doesn't render at all.
3. Change `core/home-link` to `core/loginout` in the snippet below to see a block rendering correctly as comparison.

```php
function register_home_link_block_as_navigation_last_child( $hooked_blocks, $position, $anchor_block, $context ) {
	if ( $anchor_block === 'core/navigation' && $position === 'last_child' ) {
		$hooked_blocks[] = 'core/home-link';
	}

	return $hooked_blocks;
}

add_filter( 'hooked_block_types', 'register_home_link_block_as_navigation_last_child', 10, 4 );
```

### Screenshots, screen recording, code snippet

_No response_

### Environment info

- WP 6.4
- Gutenberg latest `trunk` or 17.6 and above.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by tjcafferkey**
Fixes https://github.com/WordPress/gutenberg/issues/58307

## What?

Renders default label as "Home" rather than using `useEffect` to set the attribute in the Editor client side unless user explicitly sets it on the block themselves and saves it.

## Why?

Using the Block Hooks API it's now possible to insert this block into the Navigation without default attributes since https://github.com/WordPress/gutenberg/pull/57754.

As it stands, without a default `label` attribute this block won't render on the frontend.

## How?

This renders a default label as `Home` unless a user has set this attribute, in which case it will override the default. We are unable to set the default in `block.json` since it won't be translated.

## Testing Instructions

1. Add the below code to the `block.json` file of the Home Link block.
2. Load this in the Editor and check it's auto-inserted and the link renders as "Home".
3. Do the same for the frontend.
4. Change the text in the Editor and save.
5. Check this is reflected in the Editor upon reload, and again on the frontend.

```json
// packages/block-library/src/home-link/block.json

"blockHooks": {
	"core/navigation": "lastChild"
}
```

### CODE REVIEW

File: packages/block-library/src/home-link/edit.js [hunk 1]
```diff
@@ -64,7 +50,7 @@ export default function HomeEdit( { attributes, setAttributes, context } ) {
 					<RichText
 						identifier="label"
 						className="wp-block-home-link__label"
-						value={ label }
+						value={ label || __( 'Home' ) }
```
Comment (ockham) [inline packages/block-library/src/home-link/edit.js]: TBH I'm not sure we should change the editor-side behavior. I have a hunch that folks chose `useEffect` over a `||` fallback for a reason, which is the slightly funky UX we'd get if the user removes the label (e.g. by pressing backspace until nothing is left):

home-link

It's confusing enough that I'd rather keep the original code, even if that means that we'd allow an edge case where the user decides to remove the label, but the frontend falls back to 'Home'.

Comment (tjcafferkey) [reply packages/block-library/src/home-link/edit.js]: Hmm good catch, this must have slipped by me as I would have done "CMD + A" and started typing a new label. I think what you have suggested is a good solution, leave the editor side as-is and just make sure _something_ renders ("Home") on the server if `label` is empty. I'll update this today.

File: packages/block-library/src/home-link/index.php [hunk 2]
```diff
@@ -129,7 +129,7 @@ function block_core_home_link_build_li_wrapper_attributes( $context ) {
  */
 function render_block_core_home_link( $attributes, $content, $block ) {
 	if ( empty( $attributes['label'] ) ) {
-		return '';
+		$attributes['label'] = __( 'Home' );
```
Comment (ockham) [inline packages/block-library/src/home-link/index.php:135]: Let's maybe add a comment that this needs to be kept in sync with the `useEffect` thing in `edit.js` as I think we do elsewhere, too; and maybe another note to explain _why_ we're adding this fallback in the first place 🙂 

Something like

```suggestion
		// Using a fallback for the label attribute allows rendering the block even if no attributes have been set,
		// e.g. when using the block as a hooked block.
		// Note that the fallback value needs to be kept in sync with the one set in `edit.js` (upon first loading the block in the editor).
		$attributes['label'] = __( 'Home' );
```

(but feel free to tweak the verbiage).

Comment (tjcafferkey) [reply packages/block-library/src/home-link/index.php:135]: Yes, good point. I've accepted this suggestion.

**Review by ockham — APPROVED** [review]
Thank you!

**Pull request merged**